<?xml version="1.0" ?>
<launch>
  <!--
      Launch file for the admittance controller
  -->

  <arg name="TOPIC_ARM_STATE" default="/cartesian_velocity_controller_sim/ee_state"/>
  <arg name="TOPIC_ARM_COMMAND" default="/cartesian_velocity_controller_sim/command_cart_vel"/>
  <!-- <arg name="TOPIC_WRENCH_STATE" default="/wrench_fake"/> -->
  <arg name="TOPIC_WRENCH_STATE" default="/transformed_world"/>
  <arg name="twist_scale" default="0.7"/>


  <param name="topic_arm_state" value="$(arg TOPIC_ARM_STATE)"/>
  <param name="topic_arm_command" value="$(arg TOPIC_ARM_COMMAND)"/>
  <param name="topic_wrench_state" value="$(arg TOPIC_WRENCH_STATE)"/>
  <param name="twist_scale" value="$(arg twist_scale)"/>
  <param name="ft_arm_ready" value="false"/>

  
  <rosparam file="$(find cartesian_algorithms)/config/AdmittanceParams.yaml" command="load"/>

  <node pkg="cartesian_algorithms" name="admittance" type="Admittance_node" output="screen"/>

</launch>